content/adr/drupal-incremental-updates.md

```diff
@@ -0,0 +1,37 @@
+---
+title: "Incremental updates to Drupal"
+date: "2024-02-29"
+decision: "We will perform incremental updates to fix deprecated code in our custom modules, replacing deprecated contributed modules with latest versions or an alternative rather than one shot upgrade from one Drupal major version to another"
+status: "accepted"
+categories:
+  - Drupal
+---
+
+# Drupal 9
+
+## Context
+
+Currently, we focus on doing major security updates to contributed modules and core as part of maintenance. However, this becomes a challenge when the end of life (EOL) arrives for our current Drupal major version as it results in one large estimation for fixing deprecated custom code, replacing contributed modules, and finally upgrading the core. This causes a huge one-time cost for the clients making it challenging to get budget approvals It also is not in the spirit of smaller incremental development and hence impacts the stability of the application resulting in more efforts in each phase including QA and UAT.
```

Comment:
I have only made minor fixes but I think we need to rewrite this. It's confusing at the moment.

```suggestion
Currently, we focus on only doing major security updates to contributed modules and core as part of maintenance. However, this becomes a challenge when the end of life (EOL) arrives for our current Drupal major version as it results in one large estimation for fixing deprecated custom code, replacing contributed modules, and finally upgrading the core. This causes a huge one-time cost for the clients making it challenging to get budget approvals. It is also not in the spirit of smaller incremental development and hence impacts the stability of the application resulting in more efforts in each phase including QA and UAT.
```